### PR TITLE
Add eslint and ncc to dev deps

### PR DESCRIPTION
**Pull request by bollwyvl**
Here's a quick pass at adding and getting `eslint` to be happy. Also set up prettier to go after all the files it can, because why not.

Adding it to this repo's workflow as a fast fail should be pretty straightforward, and will save some coal burning for easily-fixable errors.

Probably won't have time to continue messing about with it today, but might already be an interesting review... i yanked most of the defaults from jupyterlab/jupyterlab, but there's room for interpretation!

### CODE REVIEW

**Comment by bollwyvl** [comment]
It appears that cross-workflow job dependencies aren't possible, or require some fairly significant investment. I guess at this point, it's quick enough to not require _too_ much optimization.

File: src/setup.ts [hunk 1]
```diff
@@ -608,17 +621,18 @@ async function setupMiniconda(
       utils.consoleLog("\n# Downloading Custom Installer...\n");
       useBundled = false;
       result = await downloadInstaller(installerUrl);
+      if (!result.ok) return result;
```
Comment (bollwyvl) [inline src/setup.ts:624]: amid all the other changes, this is actually something i missed from #43

File: .github/workflows/caching-example.yml [hunk 2]
```diff
@@ -23,7 +22,7 @@ jobs:
         with:
           path: ~/conda_pkgs_dir
           key: ${{ runner.os }}-conda-${{ env.CACHE_NUMBER }}-${{ hashFiles('etc/example-environment.yml') }}
-      - uses: goanpeca/setup-miniconda@master
+      - uses: ./
```
Comment (goanpeca) [inline .github/workflows/caching-example.yml:25]: Nice :-p 

!

**Comment by goanpeca** [comment]
>It appears that cross-workflow job dependencies aren't possible, or require some fairly significant investment. I guess at this point, it's quick enough to not require too much optimization.

Yeah, among jobs we could but not among workflows, unless we used artifact uploading, so not worth it.

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    branches:
+    - 'master'
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: 'ubuntu-latest'
+    steps:
+      - uses: actions/checkout@v2
+      - shell: bash -l {0}
```
Comment (goanpeca) [inline .github/workflows/lint.yml:17]: Should we use a specific node version and action? Cause we are now relying on what comes in the container, which could change.

Thougths?

File: .github/workflows/caching-example.yml (hunk 2 above)
Comment (bollwyvl) [reply .github/workflows/caching-example.yml:25]: yeah, that was a good one to find...

File: .github/workflows/lint.yml (hunk 3 above)
Comment (bollwyvl) [reply .github/workflows/lint.yml:17]: As an action, we kinda get `node-12` and that's that, right?

File: src/setup.ts (hunk 1 above)
Comment (goanpeca) [reply src/setup.ts:624]: Got it, I missed it too!

**Review by goanpeca — COMMENTED** [review]
Awesome addition @bollwyvl !

Left one question, otherwise LGTM!

Thanks :-)

File: .github/workflows/lint.yml (hunk 3 above)
Comment (bollwyvl) [reply .github/workflows/lint.yml:17]: Anyhow, I'm open to pinning... suggestions on what the keys/values should be?

Comment (goanpeca) [reply .github/workflows/lint.yml:17]: In theory :-p 

Let's leae like that for now, merging

**Comment by goanpeca** [comment]
Will make a 1.5 release tomorrow!

**Pull request merged**